### PR TITLE
Build gpdb with uuid library enabled

### PR DESCRIPTION
**Pull request by liuxueyang**
User wants to enable `contrib/uuid-ossp` extension. This extension
requires the uuid library.

### CODE REVIEW

**Comment by higuoxing** [comment]
Edit contrib/Makefile to enable the uuid-ossp extension.

See:
https://github.com/greenplum-db/gpdb/blob/88c511ea7f47cbb365ca886e5615d7f9df2819e0/contrib/Makefile#L14-L53

File: contrib/Makefile [hunk 1]
```diff
@@ -50,6 +50,7 @@ SUBDIRS = \
 		tsm_system_rows \
 		tsm_system_time \
 		unaccent	\
+		uuid-ossp	\
```
Comment (beeender) [inline contrib/Makefile:53]: This doesn't seem to be necessary, see https://github.com/greenplum-db/gpdb/pull/14172/files#diff-d70768358262508789165ad4e9be1292fe6e14eaabe00c4f5906a952a14ec0b1L68

```
ifneq ($(with_uuid),no)
SUBDIRS += uuid-ossp
else
ALWAYS_SUBDIRS += uuid-ossp
endif
```

Comment (beeender) [reply contrib/Makefile:53]: BTW, I didn't see any regress log on concourse, did it enable the regress tests as well?

```
REGRESS = uuid_ossp
```
https://github.com/greenplum-db/gpdb/blob/master/contrib/uuid-ossp/Makefile#L10

Comment (higuoxing) [reply contrib/Makefile:53]: Ah, seems that we need to add `--with-uuid` to more places... I didn't dig into it but found a relevant commit.

https://github.com/greenplum-db/gpdb/commit/d12fbb524d3a2db1b9fb8b2d27064e3479723cce

**Comment by liuxueyang** [comment]
- This module will be built automatically on the pipeline
- Regression test for the uuid-ossp can be run on the pipeline

File: src/test/Makefile [hunk 2]
```diff
@@ -37,6 +37,12 @@ SUBDIRS += ssl
 endif
 endif
 
+ifeq ($(with_uuid),yes)
+ifneq (,$(filter uuid-ossp,$(PG_TEST_EXTRA)))
+SUBDIRS += uuid-ossp
```
Comment (beeender) [inline src/test/Makefile:42]: I don't think this is needed. There is no subdir `uuid-ossp` in `src/test/`.

Are you sure about this change?

File: concourse/scripts/ic_gpdb.bash [hunk 3]
```diff
@@ -31,7 +31,7 @@ function gen_env(){
 		source /usr/local/greenplum-db-devel/greenplum_path.sh
 		cd "\${1}/gpdb_src"
 		source gpAux/gpdemo/gpdemo-env.sh
-		PG_TEST_EXTRA="kerberos ssl" make -s ${MAKE_TEST_COMMAND}
+		PG_TEST_EXTRA="kerberos ssl uuid-ossp" make -s ${MAKE_TEST_COMMAND}
```
Comment (beeender) [inline concourse/scripts/ic_gpdb.bash:34]: not needed I think.

**Comment by beeender** [comment]
We would also need some MPP test cases to be added.

**Review by yihong0618 — COMMENTED** [review]
LGTM

**Comment by liuxueyang** [comment]
> We would also need some MPP test cases to be added.

@beeender I have added 2 test cases: https://github.com/greenplum-db/gpdb/pull/14172/commits/638bf1e5c5d5f395c81ba2544709dc3eb8aa75ad

**Pull request merged**